SoftLayer/managers/cdn.py

```diff
@@ -5,140 +5,156 @@
-        :param string origin_url: the base URL from which content should be
-                                  pulled.
-        :param string cname: an optional CNAME that should be associated with
-                             this origin pull rule; only the hostname should be
-                             included (i.e., no 'http://', directories, etc.).
-        :param boolean secure: specifies whether this is an SSL origin pull
-                               rule, if SSL is enabled on your account
-                               (defaults to false).
+        :param str unique_id: The unique ID associated with the CDN.
+        :param dict \\*\\*kwargs: header-level option (mask)
+        :returns: The CDN object
         """
 
-        config = {'mediaType': media_type,
-                  'originUrl': origin_url,
-                  'isSecureContent': secure}
+        cdn_list = self.cdn_configuration.listDomainMappingByUniqueId(unique_id, **kwargs)
 
-        if cname:
-            config['cname'] = cname
+        # The method listDomainMappingByUniqueId() returns an array but there is only 1 object
+        return cdn_list[0]
 
-        return self.account.createOriginPullMapping(config, id=account_id)
+    def get_origins(self, unique_id, **kwargs):
+        """Retrieves list of origin pull mappings for a specified CDN account.
+
+        :param str unique_id: The unique ID associated with the CDN.
+        :param dict \\*\\*kwargs: header-level options (mask, limit, etc.)
+        :returns: The list of origin paths in the CDN object.
+        """
 
-    def remove_origin(self, account_id, origin_id):
+        return self.cdn_path.listOriginPath(unique_id, **kwargs)
+
+    def add_origin(self, unique_id, origin, path, origin_type="server", header=None,
+                   port=80, protocol='http', bucket_name=None, file_extensions=None,
+                   optimize_for="web", cache_query="include all"):
+        """Creates an origin path for an existing CDN.
+
+        :param str unique_id: The unique ID associated with the CDN.
+        :param str path: relative path to the domain provided, e.g. "/articles/video"
+        :param str origin: ip address or hostname if origin_type=server, API endpoint for
+                           your S3 object storage if origin_type=storage
+        :param str origin_type: it can be 'server' or 'storage' types.
+        :param str header: the edge server uses the host header to communicate with the origin.
+                           It defaults to hostname. (optional)
+        :param int port: the http port number (default: 80)
+        :param str protocol: the protocol of the origin (default: HTTP)
+        :param str bucket_name: name of the available resource
+        :param str file_extensions: file extensions that can be stored in the CDN, e.g. "jpg,png"
+        :param str optimize_for: performance configuration, available options: web, video, and file
```

Comment:
Is there any IBM docs that detail what these options are? If so I'd like an link to them incldued here because the details on  https://sldn.softlayer.com/reference/services/SoftLayer_Network_CdnMarketplace_Configuration_Mapping_Path/createOriginPath/ are pretty minimal.

Reply:
Done